driver-core/src/main/com/mongodb/client/model/Aggregates.java

```diff
@@ -883,6 +899,65 @@ public String toString() {
         }
     }
 
+    private static class OutStage implements Bson {
+
+        private final String collectionName;
+        private final AggregateOutStageOptions options;
+
+        OutStage(final String collectionName, final AggregateOutStageOptions options) {
+            this.collectionName = collectionName;
+            this.options = options;
+        }
+
+        @Override
+        public <TDocument> BsonDocument toBsonDocument(final Class<TDocument> tDocumentClass, final CodecRegistry codecRegistry) {
+            BsonDocumentWriter writer = new BsonDocumentWriter(new BsonDocument());
+            writer.writeStartDocument();
+
+            writer.writeName("$out");
+            if (optionsAreAllDefault()) {
+                writer.writeString(collectionName);
+            } else {
+                writer.writeStartDocument();
+                writer.writeName("mode");
+                switch (options.getMode()) {
+                    case REPLACE_COLLECTION:
+                        writer.writeString("replaceCollection");
+                        break;
+                    case REPLACE_DOCUMENTS:
+                        writer.writeString("replaceDocuments");
+                        break;
+                    case INSERT_DOCUMENTS:
+                        writer.writeString("insertDocuments");
+                        break;
+                    default:
+                        throw new IllegalArgumentException("Unsupported mode: " + options.getMode());
+                }
+                writer.writeString("to", collectionName);
+
+                String databaseName = options.getDatabaseName();
+                if (databaseName != null) {
+                    writer.writeString("db", databaseName);
+                }
+
+                Bson uniqueKey = options.getUniqueKey();
+                if (uniqueKey != null) {
+                    writer.writeName("uniqueKey");
+                    BuildersHelper.encodeValue(writer, uniqueKey, codecRegistry);
+                }
+
+                writer.writeEndDocument();
+            }
+
+            writer.writeEndDocument();
+            return writer.getDocument();
+        }
+
+        private boolean optionsAreAllDefault() {
```

Comment:
I debated whether to put this method on the options class but decided against it since it's only the builder that cares.